src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -40,29 +34,60 @@ import org.gradle.kotlin.dsl.withType
 @Suppress("UnstableApiUsage")
 class NexusPublishPlugin : Plugin<Project> {
 
-    companion object {
-        private val serverUrlToStagingRepoUrl = ConcurrentHashMap<URI, URI>()
-    }
-
     override fun apply(project: Project) {
-        project.pluginManager.apply(MavenPublishPlugin::class)
+        require(project == project.rootProject) {
+            "Plugin must be applied to the root project"
+        }
 
-        project.gradle.addBuildListener(object : BuildAdapter() {
-            override fun buildFinished(result: BuildResult) {
-                serverUrlToStagingRepoUrl.clear()
+        val extension = project.extensions.create<NexusPublishExtension>(NexusPublishExtension.NAME, project)
+        configureTasks(project, extension)
+        configureTaskDependenciesForAllProjects(project, extension)
+
+        project.rootProject.plugins.withId("io.codearte.nexus-staging") {
+            val nexusStagingExtension = project.rootProject.the<NexusStagingExtension>()
```

Comment:
should this be removed?

Reply:
Yes, that was done as part of #24, I've rebased onto `master` now.